express_relay/sdk/python/express_relay_utils/express_relay_types.py

```diff
@@ -0,0 +1,318 @@
+        else:
+            assert self.result is None, "result must be None"
+        return self
+
+
+class OpportunityBid(BaseModel):
+    """
+    Attributes:
+        opportunity_id: The ID of the opportunity.
+        amount: The amount of the bid in wei.
+        executor: The address of the executor.
+        permission_key: The permission key to bid on.
+        signature: The signature of the bid.
+        valid_until: The unix timestamp after which the bid becomes invalid.
+    """
+
+    opportunity_id: UUIDString
+    amount: IntString
+    executor: Address
+    permission_key: HexString
+    signature: SignedMessageString
+    valid_until: IntString
+
+    model_config = {
+        "arbitrary_types_allowed": True,
+    }
+
+
+class OpportunityParamsV1(BaseModel):
+    """
+    Attributes:
+        target_calldata: The calldata for the contract call.
+        chain_id: The chain ID to bid on.
+        target_contract: The contract address to call.
+        permission_key: The permission key to bid on.
+        buy_tokens: The tokens to receive in the opportunity.
+        sell_tokens: The tokens to spend in the opportunity.
+        target_call_value: The value to send with the contract call.
+        version: The version of the opportunity.
+    """
+
+    target_calldata: HexString
+    chain_id: str
+    target_contract: Address
+    permission_key: HexString
+    buy_tokens: list[TokenAmount]
+    sell_tokens: list[TokenAmount]
+    target_call_value: IntString
+    version: Literal["v1"]
+
+
+class OpportunityParams(BaseModel):
+    """
+    Attributes:
+        params: The parameters of the opportunity.
+    """
+
+    params: Union[OpportunityParamsV1] = Field(..., discriminator="version")
+
+
+class Opportunity(BaseModel):
+    """
+    Attributes:
+        target_calldata: The calldata for the contract call.
+        chain_id: The chain ID to bid on.
+        target_contract: The contract address to call.
+        permission_key: The permission key to bid on.
+        buy_tokens: The tokens to receive in the opportunity.
+        sell_tokens: The tokens to spend in the opportunity.
+        target_call_value: The value to send with the contract call.
+        version: The version of the opportunity.
+        creation_time: The creation time of the opportunity.
+        opportunity_id: The ID of the opportunity.
+    """
+
+    target_calldata: HexString
+    chain_id: str
+    target_contract: Address
+    permission_key: HexString
+    buy_tokens: list[TokenAmount]
+    sell_tokens: list[TokenAmount]
+    target_call_value: IntString
+    version: str
+    creation_time: IntString
+    opportunity_id: UUIDString
+
+    supported_versions: ClassVar[list[str]] = ["v1"]
+    unsupported_version_error_text: ClassVar[str] = (
+        "Cannot handle opportunity version: {%s}. Please upgrade your client."
+    )
+
+    @model_validator(mode="after")
```

Comment:
I think we should do this before the validation. If the new schema doesn't have some of these fields we will receive another exception which we will not handle